src/components/DialpadPanel.vue

```diff
@@ -0,0 +1,228 @@
+<!--
+  - @copyright Copyright (c) 2023 Maksim Sukharev <antreesy.web@gmail.com>
+  -
+  - @author Maksim Sukharev <antreesy.web@gmail.com>
+  -
+  - @license AGPL-3.0-or-later
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<template>
+	<NcPopover id="dial-popover"
+		ref="popover"
+		popover-base-class="dial-popover"
+		:container="container"
+		@show="prefillCode">
+		<template #trigger>
+			<NcButton>
+				<template #icon>
+					<Dialpad :size="20" />
+				</template>
+			</NcButton>
+		</template>
+
+		<div ref="panel"
+			class="dial-panel"
+			tabindex="0"
+			@keydown.capture="handleKeyDown">
+			<NcSelect class="dial-panel__select"
+				:value="region"
+				:options="options"
+				:append-to-body="false"
+				:clearable="false"
+				:placeholder="t('spreed', 'Select a region')"
+				label="dial_and_name"
+				@option:selected="dialCode">
+				<template #option="{ emoji, dial_code, name }">
+					<div class="dial-panel__select-option">
+						<p>{{ emoji }} {{ dial_code }}</p>
+						<em>{{ name }}</em>
+					</div>
+				</template>
+				<template #selected-option="{ emoji, dial_code, name }">
+					<span class="dial-panel__select-option">
+						<b>{{ emoji }} {{ dial_code }}</b> {{ name }}
+					</span>
+				</template>
+			</NcSelect>
+
+			<NcButton v-for="button in buttons"
+				:key="button.value"
+				class="dial-panel__button"
+				@click="dialType(button.value)">
+				{{ button.label }}
+			</NcButton>
+		</div>
+	</NcPopover>
+</template>
+
+<script>
+import { getLocale } from '@nextcloud/l10n'
+
+import NcButton from '@nextcloud/vue/dist/Components/NcButton.js'
+import NcPopover from '@nextcloud/vue/dist/Components/NcPopover.js'
+import NcSelect from '@nextcloud/vue/dist/Components/NcSelect.js'
+
+import Dialpad from 'vue-material-design-icons/Dialpad.vue'
+
+import { regionCodes } from '../regionCodes.js'
+
+const allowedKeys = /^[0-9+\-()*#\s]+$/
+
+const buttons = [
+	{ value: '1', label: '1', },
+	{ value: '2', label: '2', },
+	{ value: '3', label: '3', },
+	{ value: '4', label: '4', },
+	{ value: '5', label: '5', },
+	{ value: '6', label: '6', },
+	{ value: '7', label: '7', },
+	{ value: '8', label: '8', },
+	{ value: '9', label: '9', },
+	{ value: '*', label: '*', },
+	{ value: '0', label: '0', },
+	{ value: '#', label: '#', },
```

Comment:
There will be two sets introduced, for adding number, and for dialing in call

![image](https://github.com/nextcloud/spreed/assets/93392545/5bf9d77a-710c-49bc-aa22-9a91489ea379) ![image](https://github.com/nextcloud/spreed/assets/93392545/7bf8a7c1-7f5d-40ac-ae88-0ee976349ae2)

Reply:
I know, I am referring to dialing in call. Asterik and hash symbols are usually used to dial your voicemail or *69 to call the last incoming call etc...  These functions are not available , right ? 
If so, numbers / backspace should be enough for basic phone calling.

Reply:
They should be available depending on a dial provider, so some companies allow to use * in # for some specific cases.
I'd keep them anyway, because we don't know beforehand, which configuration was provided to the customers by their local dial providers

Reply:
The symbols `*` and `#` are useful when sending DTMF tones while already connected by phone. Often, `#` is used to signal the end of a entered number (e.g. when entering the room number / participant pin when dialing *into* Talk). `*` is often used to start a command by phone (e.g. `*6` to mute the own phone).